src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -406,91 +388,7 @@ protected override void Free()
-            {
-                _hasInitialState = true;
-                var variableBySlot = initialState.VariableBySlot;
-                nextVariableSlot = variableBySlot.Length;
-                foreach (var (variable, slot) in initialState.VariableSlot)
-                {
-                    Debug.Assert(slot < nextVariableSlot);
-                    _variableSlot.Add(variable, slot);
-                }
-                this.variableBySlot = variableBySlot.ToArray();
-                foreach (var (key, value) in initialState.VariableTypes)
-                {
-                    _variableTypes.Add(key, value);
-                }
-                this.State = initialState.VariableNullableStates.Clone();
-            }
-            else
-            {
-                _hasInitialState = false;
-            }
-        }
-
-        internal sealed class IsSimpleMethodVisitor : BoundTreeWalkerWithStackGuard
-        {
-            private bool _hasComplexity;
-
-            internal static bool IsSimpleMethod(BoundNode? node)
-            {
-                if (node is BoundConstructorMethodBody constructorBody && constructorBody.Initializer is { })
-                {
-                    return false;
-                }
-                if (node is BoundMethodBodyBase methodBody)
-                {
-                    var blockBody = methodBody.BlockBody;
-                    var expressionBody = methodBody.ExpressionBody;
-                    node = blockBody;
-                    if (node is { })
-                    {
-                        if (expressionBody is { }) return false;
-                    }
-                    else
-                    {
-                        node = expressionBody;
-                    }
-                }
-                var visitor = new IsSimpleMethodVisitor();
-                try
-                {
-                    visitor.Visit(node);
-                    return !visitor._hasComplexity;
-                }
-                catch (CancelledByStackGuardException)
-                {
-                    return false;
-                }
-            }
-
-            public override BoundNode? Visit(BoundNode? node)
-            {
-                if (node is null)
-                {
-                    return null;
-                }
-                if (_hasComplexity)
-                {
-                    return node;
-                }
-                if (node is BoundExpression)
-                {
-                    return base.Visit(node);
-                }
-                switch (node.Kind)
-                {
-                    case BoundKind.Block:
-                    case BoundKind.ExpressionStatement:
-                    case BoundKind.ReturnStatement:
-                        return base.Visit(node);
-                }
-                _hasComplexity = true;
-                return node;
-            }
+            _hasInitialState = variables is { };
```

Comment:
>{ } [](start = 44, length = 3)

Consider using `not null` for clarity.

Reply:
I prefer this syntax since it's shorter.

---
In reply to: [560538257](https://github.com/dotnet/roslyn/pull/50417#discussion_r560538257) [](ancestors = 560538257)

Reply:
@jaredpar what have you done.

---
In reply to: [560598649](https://github.com/dotnet/roslyn/pull/50417#discussion_r560598649) [](ancestors = 560598649,560538257)